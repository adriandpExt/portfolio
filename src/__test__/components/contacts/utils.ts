import { fireEvent } from "@testing-library/react";

export const changeInputValue = (
  container: HTMLElement,
  testId: string,
  value: string
) => {
  const inputElement = container.querySelector(
    `[data-testid="${testId}"] input`
  );
  if (!inputElement) {
    return new Error(`Input element with data-testid "${testId}" not found`);
  }
  fireEvent.change(inputElement, { target: { value } });
};

export const changeTextareaValue = (
  container: HTMLElement,
  testId: string,
  value: string
) => {
  const inputElement = container.querySelector(
    `[data-testid="${testId}"] textarea`
  );
  if (!inputElement) {
    return new Error(`Input element with data-testid "${testId}" not found`);
  }
  fireEvent.change(inputElement, { target: { value } });
};
