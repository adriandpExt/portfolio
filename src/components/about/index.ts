export * from "./About";
export * from "./Wrapper";
