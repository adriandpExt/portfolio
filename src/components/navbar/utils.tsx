import AssignmentIndIcon from "@mui/icons-material/AssignmentInd";
import HomeIcon from "@mui/icons-material/Home";
import AppsIcon from "@mui/icons-material/Apps";
import ContactMailIcon from "@mui/icons-material/ContactMail";
import InfoIcon from "@mui/icons-material/Info";

export const menuItems = [
  { listIcon: <HomeIcon />, listText: "Home", listPath: "/portfolio " },
  {
    listIcon: <AssignmentIndIcon />,
    listText: "Resume",
    listPath: "/portfolio/cv",
  },
  {
    listIcon: <InfoIcon />,
    listText: "About",
    listPath: "/portfolio/about",
  },
  {
    listIcon: <AppsIcon />,
    listText: "Projects",
    listPath: "/portfolio/project",
  },
  {
    listIcon: <ContactMailIcon />,
    listText: "Contact",
    listPath: "/portfolio/contact",
  },
];
