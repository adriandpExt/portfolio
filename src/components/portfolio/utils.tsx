// import avatar from "../../assets/image/avatar.png";
import empf1 from "../../assets/empf/empf1.png";
import empf2 from "../../assets/empf/empf2.png";
import empf3 from "../../assets/empf/empf3.png";

export const projects = [
  {
    name: "EMPF",
    description: `EMPF is a centralised electronic platform that aims to standardise and automate the administration processes of MPF schemes. An electronic platform to be launched by the Mandatory Provident Fund Schemes Authority (MPFA), it will cover all 12 MPF trustees, as well as the 357,000 employers and 4.7 million members that Hong Kong's mandatory pension scheme covers. Collaboration with other developer`,
    image: [empf1, empf2, empf3],
  },
];
