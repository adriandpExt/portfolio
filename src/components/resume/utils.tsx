export const details = [
  {
    jobtitle: "Solutions Developer",
    companyName: "LENOVO PCCW SOLUTIONS PHILIPPINES INC",
    year: "2022-2024",
    description: `Motivated Solution Developer with a dedication to delivering sophisticated and effective web applications. I am in search of a challenging position where I can utilize my two years of expertise in front-end development and UI design, employing contemporary technologies to contribute to the company's success`,
  },
];
