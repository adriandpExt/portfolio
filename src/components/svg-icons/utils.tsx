export type IconName =
  | "ic_gmail"
  | "ic_mobile"
  | "ic_facebook"
  | "ic_github"
  | "ic_linkedin"
  | "ic_vue"
  | "ic_react"
  | "ic_html"
  | "ic_css"
  | "ic_javascript"
  | "ic_bootstrap"
  | "ic_material_ui"
  | "ic_tailwind";
