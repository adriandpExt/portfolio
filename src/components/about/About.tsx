import { ReactNode } from "react";
import { motion } from "framer-motion";

import Box from "@mui/material/Box";
import Grid from "@mui/material/Grid";
import Typography from "@mui/material/Typography";

import me from "../../assets/image/me_id.png";

import { SvgIcons } from "../svg-icons";

import Wrapper from "./Wrapper";

import { techStack, techTools } from "./utils";
import { Stack } from "@mui/material";

export const About = () => {
  const renderImage = () => {
    return (
      <Box>
        <motion.img
          src={me}
          alt="me"
          height="50%"
          width="70%"
          style={{
            filter: "grayscale(100%) drop-shadow(15px 10px 4px crimson)",
          }}
          whileHover={{
            filter: "grayscale(0%) drop-shadow(15px 10px 4px tomato)",
            scale: 1.5,
          }}
          transition={{ duration: 0.5 }}
        />
      </Box>
    );
  };

  const renderStack = (): ReactNode => {
    return (
      <Grid container spacing={1}>
        {techStack.map((item, id) => (
          <Grid item xs={6} sm={4} md={3} lg={2} key={id}>
            <Wrapper>
              <SvgIcons
                name={item}
                width={50}
                height={50}
                data-testid="tech-stack-icon"
              />
            </Wrapper>
          </Grid>
        ))}
      </Grid>
    );
  };

  const renderTools = (): ReactNode => {
    return (
      <Grid container spacing={1}>
        {techTools.map((item, id) => (
          <Grid item xs={6} sm={4} md={3} lg={2} key={id}>
            <Wrapper>
              <SvgIcons
                name={item}
                width={50}
                height={50}
                data-testid="tech-tools-icon"
              />
            </Wrapper>
          </Grid>
        ))}
      </Grid>
    );
  };

  const renderAboutIntro = () => {
    return (
      <Stack gap={3}>
        <Typography color="#FFF" variant="h6" sx={{ fontFamily: "cursive" }}>
          Hello! I'm{" "}
          <span
            style={{ color: "tomato", fontSize: "2rem", fontStyle: "italic" }}
          >
            Adrian Del Prado
          </span>
          , and I like learning things and building them.
        </Typography>

        <Typography color="#FFF" variant="h6" sx={{ fontFamily: "cursive" }}>
          The reason why I chose web development was that there was a lot to
          learn and a lot of things to build.
        </Typography>

        <Typography color="#FFF" variant="h6" sx={{ fontFamily: "cursive" }}>
          I am fluent in classics like{" "}
          <span
            style={{ color: "tomato", fontSize: "2rem", fontStyle: "italic" }}
          >
            {" "}
            React JS, Typescript{" "}
          </span>{" "}
          And{" "}
          <span
            style={{ color: "tomato", fontSize: "2rem", fontStyle: "italic" }}
          >
            {" "}
            React Frameworks
          </span>
        </Typography>

        <Typography color="#FFF" variant="h6" sx={{ fontFamily: "cursive" }}>
          My field of Interest's are building new{" "}
          <span
            style={{ color: "tomato", fontSize: "2rem", fontStyle: "italic" }}
          >
            Web Technologies and Products
          </span>
        </Typography>
      </Stack>
    );
  };

  return (
    <Grid
      container
      rowGap={5}
      sx={{
        justifyContent: "center",
        alignItems: "center",
        padding: "2rem",
        alignContent: "center",
        alignSelf: "center",
        paddingX: "5rem",
        whiteSpace: "wrap",
      }}
    >
      <Grid item xs={12} sm={9}>
        {renderAboutIntro()}
      </Grid>

      <Grid item xs={12} sm={3}>
        {renderImage()}
      </Grid>

      <Grid item xs={12}>
        <Typography color="#FFF" variant="h5" sx={{ fontFamily: "cursive" }}>
          Tech Stack
        </Typography>
        {renderStack()}
      </Grid>

      <Grid item xs={12}>
        <Typography color="#FFF" variant="h5" sx={{ fontFamily: "cursive" }}>
          Tech Tools
        </Typography>
        {renderTools()}
      </Grid>
    </Grid>
  );
};

export default About;
