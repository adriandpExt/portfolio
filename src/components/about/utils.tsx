import { IconName } from "../svg-icons/utils";

export const techStack: IconName[] = [
  "ic_react",
  "ic_vue",
  "ic_javascript",
  "ic_typescript",
  "ic_html",
  "ic_css",
  "ic_bootstrap",
  "ic_tailwind",
  "ic_material_ui",
];
export const techTools: IconName[] = ["ic_git", "ic_mysql", "ic_vscode"];
