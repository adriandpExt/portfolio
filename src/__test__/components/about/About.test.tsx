import "@testing-library/jest-dom";

import { render, fireEvent, screen } from "@testing-library/react";

import About from "../../../components/about/About";

import { techStack, techTools } from "../../../components/about/utils";

describe("About Component Tests", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("Render Test", () => {
    const { container } = render(<About />);
    expect(container).toMatchSnapshot();
  });

  test("CV Download Test", () => {
    render(<About />);

    const downloadButton = screen.getByText("CV DOWNLOAD");

    fireEvent.click(downloadButton);
    expect(downloadButton).toHaveAttribute(
      "href",
      "public/cv/AdrianDelPrado_CV.pdf"
    );
    expect(downloadButton).toHaveAttribute(
      "download",
      "adrian_del_prado_cv.pdf"
    );
    expect(downloadButton).toHaveAttribute("target", "_blank");
  });

  test("Tech Stack Render Test", () => {
    render(<About />);
    const stackIcons = screen.getAllByTestId("tech-stack-icon");
    expect(stackIcons.length).toBe(techStack.length);
  });

  test("Tech Tools Render Test", () => {
    render(<About />);
    const toolsIcons = screen.getAllByTestId("tech-tools-icon");
    expect(toolsIcons.length).toBe(techTools.length);
  });

  test("Image Hover Test", () => {
    render(<About />);
    const image = screen.getByAltText("me");

    fireEvent.mouseOut(image);
    expect(image).toHaveStyle(
      "filter: grayscale(100%) drop-shadow(15px 10px 4px crimson)"
    );
  });
});
