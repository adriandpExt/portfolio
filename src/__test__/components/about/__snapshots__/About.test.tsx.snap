// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component Tests Render Test 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-dh6qnx-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9 css-173qa7i-MuiGrid-root"
    >
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-15avm3p-MuiButtonBase-root-MuiButton-root"
        download="adrian_del_prado_cv.pdf"
        href="https://drive.google.com/file/d/1f4PxPAWm7zZONOFaGRqayO5V4cDpBlTX/view?usp=drive_link"
        tabindex="0"
        target="_blank"
      >
        CV DOWNLOAD
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <div
        class="MuiStack-root css-msk5bl-MuiStack-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-mivlh5-MuiTypography-root"
        >
          Hello! I'm
           
          <span
            style="color: tomato; font-size: 2rem; font-style: italic;"
          >
            Adrian Del Prado
          </span>
          , and I like learning things and building them.
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-mivlh5-MuiTypography-root"
        >
          The reason why I chose web development was that there was a lot to learn and a lot of things to build.
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-mivlh5-MuiTypography-root"
        >
          I am fluent in classics like
           
          <span
            style="color: tomato; font-size: 2rem; font-style: italic;"
          >
             
            React JS, Typescript
             
          </span>
           
          And
           
          <span
            style="color: tomato; font-size: 2rem; font-style: italic;"
          >
             
            React Frameworks
          </span>
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-mivlh5-MuiTypography-root"
        >
          My field of Interest's are building new
           
          <span
            style="color: tomato; font-size: 2rem; font-style: italic;"
          >
            Web Technologies and Products
          </span>
        </h6>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3 css-zbaamk-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt="me"
          height="50%"
          src="[object Object]"
          style="filter: grayscale(100%) drop-shadow(15px 10px 4px crimson);"
          width="70%"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-k4zp2n-MuiTypography-root"
      >
        Tech Stack
      </h5>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-stack-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_react"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-stack-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_vue"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-stack-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_javascript"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-stack-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_typescript"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-stack-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_html"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-stack-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_css"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-stack-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_bootstrap"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-stack-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_tailwind"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-stack-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_material_ui"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-k4zp2n-MuiTypography-root"
      >
        Tech Tools
      </h5>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-tools-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_git"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-tools-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_mysql"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1pgzwh3-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-f72d4r"
          >
            <div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gwklqv-MuiSvgIcon-root"
                  data-testid="tech-tools-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-ic_vscode"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
